interpreter/interpreter.go

```diff
@@ -151,6 +156,185 @@ func CompileRegexConstants(regexOptimizations ...*RegexOptimization) Interpretab
 	return decRegexOptimizer(regexOptimizations...)
 }
 
+type typeVerifier func(int64, ...*types.TypeValue) (bool, error)
+
+// CheckStringFormat checks the syntax and cardinality of any string.format calls present in the expression and reports
+// any errors at compile time.
+func CheckStringFormat(enableCheckArgTypes bool, verifier typeVerifier) InterpretableDecorator {
```

Comment:
Split out some utility functions here? The size and depth of nesting in this function harms readability.

Reply:
I've moved out the datatype-specific checks into their own separate functions, so the layout now is fairly close to what `string.format` has. After the duplicated lexer functionality is removed it should be a bit cleaner, but I'll wait to resolve this until after that refactor is pushed.